lupa/_lupa.pyx

```diff
@@ -1554,15 +1532,22 @@ cdef object execute_lua_call(LuaRuntime runtime, lua_State *L, Py_ssize_t nargs)
 
 cdef int push_lua_arguments(LuaRuntime runtime, lua_State *L,
                             tuple args, bint first_may_be_nil=True) except -1:
-    cdef int i
+    cdef int i, n
+    cdef Py_ssize_t nargs
+    cdef bint wrap_none = not first_may_be_nil
     if args:
-        old_top = lua.lua_gettop(L)
-        for i, arg in enumerate(args):
-            if not py_to_lua(runtime, L, arg, wrap_none=not first_may_be_nil):
-                lua.lua_settop(L, old_top)
-                raise TypeError("failed to convert argument at index %d" % i)
-            first_may_be_nil = True
-    return 0
+        nargs = len(args)
+        if nargs > INT_MAX:
```

Comment:
This might give a C compiler warning on systems where `sizeof(int)` == `sizeof(Py_ssize_t)`, since the condition would always be false there.

Reply:
Wouldn't it be simply optimized out?

Reply:
Probably, yes, after giving the warning.